plugin/__tests__/fixtures/buildGradleFiles.ts

```diff
@@ -0,0 +1,164 @@
+const appBuildGradleWithoutIAP = `
+apply plugin: "com.android.application"
+
+import com.android.build.OutputFile
+
+def reactNativeArchitectures() {
+    def value = project.getProperties().get("reactNativeArchitectures")
+    return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
+}
+
+android {
+    ndkVersion rootProject.ext.ndkVersion
+
+    compileSdkVersion rootProject.ext.compileSdkVersion
+
+    defaultConfig {
+        applicationId 'com.test.withIAP'
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.targetSdkVersion
+        versionCode 34
+        versionName "1.16.2"
+        buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()`;
+
+const appBuildGradleWithPlayStoreIAP = `
+apply plugin: "com.android.application"
+
+import com.android.build.OutputFile
+
+def reactNativeArchitectures() {
+    def value = project.getProperties().get("reactNativeArchitectures")
+    return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
+}
+
+android {
+    ndkVersion rootProject.ext.ndkVersion
+
+    compileSdkVersion rootProject.ext.compileSdkVersion
+
+    defaultConfig {
+missingDimensionStrategy "store", "play"
```

Comment:
Looks like these variables have some duplication, any thoughts on using a common template?

Reply:
Agree that there is too much of it. In current unit tests implementation, we are passing initial `build.gradle` (part of it to be precise) to the function and comparing the result with the correct version. This is done 3 times, one for each `paymentProvider`. We could remove irrelevant lines and leave only lines close to `defaultConfig`. What do you think?